folium/vector_layers.py

```diff
@@ -322,8 +322,8 @@ class CircleMarker(Marker):
             {% endmacro %}
             """)
 
-    def __init__(self, location, radius=10, popup=None, tooltip=None, **kwargs):
+    def __init__(self, location=None, radius=10, fill=True, popup=None, tooltip=None, **kwargs):
```

Comment:
Why are you adding `fill` here explicitly? It's currently parsed from the `**kwargs` in `_parse_options()`, so no need to add it explicitely. Also, the default is currently not `True` but dependent on the `fill_color`. In `_parse_options()`, if `fill` is not provided it is set to `True` if there is a `fill_color` and `False` otherwise. So it can be used as `kwarg`.

Reply:
My thought is that these would be a good opinionated default value for these Vector Marker layers.

I think because when I was first starting to use the API, this tripped me up several times as I wasn't yet familiar with Leaflet, and even when I passed a `color` argument, I could not get my popup/hover methods to display except in the outer radius - and I'm sure that other people have had similar experiences, as there is no documentation of the `fill`, `fill_color`, or `color` arguments in the docstring to help users make sense of how to use the class - they have to go to the Leaflet docs. (I probably could/should have added some annotations in the docstring. Maybe I should make another PR for this).

If a user is adding a marker to the Map, I think it's reasonable to assume they are interested in the interior of the vector and would like click/hover events to fire with the mouse inside of the marker, which will only happen if `fill` or `fill_color` has a value. I don't think we should pass a default `fill_color`, but I think `fill=True` makes sense unless the user really wants to set it `False`, in which case they still have the option. `fill=True` still leaves the vector interior transparent.

To me this is in the same vein higher level ease of use that folium does with many other things, like `tiles="OpenStreetMap"` in `Map`.

Let me know what you think.

Reply:
That makes sense, thanks for explaining! Maybe a tip for making our lives easier: this is really something for its own PR. You show the change (`fill=True`), explain why (which you did nicely here), and then it can be merged quickly without much reviewing.